doc/controller.md

```diff
@@ -0,0 +1,132 @@
+# A case for an etcd controller
+
+## Abstract
+
+This document tries to show that introducing a Kubernetes controller would increase the options for solving the various problems encountered in the life-cycle management of `etcd` instances.
+
+## Goal
+
+To justify the introduction of a Kubernetes controller into the life-cycle management of `etcd` instances in terms of the increased number of options to solve the known problems in the domain.
+
+## Non-goal
+
+It is not a goal of this document to recommend rewriting of all the current functionality in etcd-backup-restore as a Kubernetes [operator](https://coreos.com/operators/).
+The options highlighted below are just that. Options. 
+
+Even if there is a consensus on adopting the controller way to implement some subset of the functionality below, it is not the intention of this document to advocate for implementing all those features right away.
+
+## Content
+
+* [A case for an etcd controller](#a-case-for-an-etcd-controller)
+  * [Abstract](#abstract)
+  * [Goal](#goal)
+  * [Non\-goal](#non-goal)
+  * [Content](#content)
+  * [Single\-node and Multi\-node](#single-node-and-multi-node)
+  * [Database Restoration](#database-restoration)
+  * [Database Incremental Backup Compaction](#database-incremental-backup-compaction)
+  * [Database Verification](#database-verification)
+  * [Autoscaling](#autoscaling)
+  * [Non\-disruptive Maintenance](#non-disruptive-maintenance)
+  * [Co\-ordination outside of Gardener Reconciliation](#co-ordination-outside-of-gardener-reconciliation)
+  * [Backup Health Verification](#backup-health-verification)
+  * [Major Version Upgrades](#major-version-upgrades)
+  * [Summary](#summary)
+
+## Single-node and Multi-node
+
+We currently support only single-node `etcd` instances.
+There might be valid reasons to also support multi-node `etcd` clusters including the non-disruptive maintenance and support for Gardener Ring.
+
+A multi-node `etcd` cluster requires co-ordination between the `etcd` nodes not just for consensus management but also for life-cycle management tasks.
+This means that co-ordination is required across nodes for some (or all) tasks such as backups (full and delta), verification, restoration and scaling.
+Some of that co-ordination could be done with the current sidecar approach.
+But many of them, especially, restoration and scaling are done better with a controller.
+
+For backups, a better option might be to separate the backup sidecar as a separate pod for the whole etcd cluster rather than a sidecar container for each etcd instance in the etcd cluster.
+This could be optionally used only in the multi-node scenario and we can perhaps continue to use the sidecar container approach for the single-node scenario.
+This could be the first step and we can think of introducing the controller for co-ordination for other life-cycle operations later.
+
```

Comment:
Yes, multi-node is definitely something that will require a coordinator/controller. So far I see three use cases: 
-	Cross-cluster Gardener (whether it’s a self-managed ring is a separate topic, but not relevant for this question)
-	Smooth rolling update with hand-over between two etcd instances (what you call shorter/better just “non-disruptive maintenance”)
-	Multi-zonal control planes (desired from customer PoV, but less so from technical/our PoV, so with the lowest priority of the three at present)